### PR TITLE
implemented chroma subsampling

### PR DESCRIPTION
**Pull request by jesi-rgb**
Implemented chroma subsampling under the ImageUtils scene as the function `chroma_subsample_image`.

Also, edited `get_yuv_image_from_rgb` to include a `mapped` attribute, which enables returning back the mapped image or the raw data, the latter being necessary for the chroma subsampling algorithm.

The algorithm takes a pixel array and returns the same image but chroma subsampled.

A test scene `IntroChromaSubsampling` is created to show how the system works. The scene presents three versions of the Shed.jpg file: the raw one, chroma subsampled and a difference image to certify that the images are different. The end result looks so close to the original that I was doubting if I performed it correctly.

### CODE REVIEW

File: JPEGImageCompression/image_compression.py [hunk 1]
```diff
@@ -203,22 +210,93 @@ def get_yuv_image_from_rgb(self, pixel_array):
...
+
+            # then the v channel
+            for i in range(0, v.shape[0], 2):
+                out_v[i : i + 2] = np.mean(v[i : i + 2], axis=0)
+
+        # Downsample with a window of 2 in both directions
+        elif mode == "4:2:0":
+            for i in range(0, u.shape[0], 2):
+                for j in range(0, u.shape[1], 2):
+                    out_u[i : i + 2, j : j + 2] = np.mean(u[i : i + 2, j : j + 2])
```
Comment (nipunramk) [inline JPEGImageCompression/image_compression.py:273]: Minor nit: stylistically, when slicing indices of a list/array, make sure there are no spaces in between any characters.

E.g here, it should be `out_u[i:i+2, j:j+2] = np.mean(u[i:i+2, j:j+2])` -- this seems to be the accepted standard for style guidelines of slicing -- https://stackoverflow.com/questions/8092513/style-formatting-the-slice-operator

Comment (jesi-rgb) [reply JPEGImageCompression/image_compression.py:273]: Gotcha!

Comment (jesi-rgb) [reply JPEGImageCompression/image_compression.py:273]: Actually, looking a bit more into it, seems like black (my formatter) has already had some issues that have been resolved regarding this topic.

Acording to [PEP-8](https://www.python.org/dev/peps/pep-0008/#:~:text=%23%20Correct%3A%0Aham%5B1%3A9%5D%2C%20ham%5B1%3A9%3A3%5D%2C%20ham%5B%3A9%3A3%5D%2C%20ham%5B1%3A%3A3%5D%2C%20ham%5B1%3A9%3A%5D%0Aham%5Blower%3Aupper%5D%2C%20ham%5Blower%3Aupper%3A%5D%2C%20ham%5Blower%3A%3Astep%5D%0Aham%5Blower%2Boffset%20%3A%20upper%2Boffset%5D%0Aham%5B%3A%20upper_fn(x)%20%3A%20step_fn(x)%5D%2C%20ham%5B%3A%3A%20step_fn(x)%5D%0Aham%5Blower%20%2B%20offset%20%3A%20upper%20%2B%20offset%5D), the formatting I used is indeed compliant with the style guides.

Comment (nipunramk) [reply JPEGImageCompression/image_compression.py:273]: Got it, ok , I think it fine as is then! You may need to do a minor rebase on the code -- I pushed some DCT and JPEG related changes and seems like there is one small conflict with your change.

**Comment by jesi-rgb** [comment]
I think everything is in order now! Ready to merge

**Review by nipunramk — APPROVED** [review]

**Pull request merged**